pkg/backend/filestate/lock.go

```diff
@@ -79,6 +79,36 @@ func (b *localBackend) checkForLock(ctx context.Context, stackRef backend.StackR
 		}
 	}
 
+	return lockKeys, nil
+}
+
+// checkForLock looks for any existing locks for this stack, and returns a helpful diagnostic if there is one.
+func (b *localBackend) checkForLock(
+	ctx context.Context,
+	stackRef backend.StackReference,
+	lockTimeout time.Duration,
+) error {
+	lockKeys, err := b.getLockKeys(ctx, stackRef)
+	if err != nil {
+		return err
+	}
+
+	start := time.Now()
+
+	for len(lockKeys) > 0 && time.Since(start) < lockTimeout {
+		timeLeft := lockTimeout - time.Since(start)
+		b.d.Infof(diag.Message("", fmt.Sprintf("State is locked, waiting for lock... %s left", timeLeft.Truncate(time.Second))))
```

Comment:
We should probably skip some of these messages otherwise a fast check loop is gonna be _very_ spammy.
If we print this at most once a second that would probably be fine.

Reply:
Do you have a suggestion for how to accomplish that? I could do a `time.Sleep` but that feels... hacky

Reply:
Just track when you last wrote the log, then if time.Since for that timestamp is >1s write the log and update that timestamp.

Reply:
Done